dspy/retrieve/MyScaleRM.py

```diff
@@ -128,8 +124,14 @@ def setup_local_model(self, model_name: str):
             self.device = torch.device('cpu')
 
         self._local_embed_model.to(self.device)
-
+    
     def get_embeddings(self, queries: List[str]) -> List[List[float]]:
+        if cache_turn_on:
```

Comment:
this is close, but I was thinking something along the lines of what's done in [dspy.OpenAI](https://github.com/stanfordnlp/dspy/blob/f2a96137288925be7a9ad46b5c40c1e0425bd531/dsp/modules/gpt3.py#L218) for openai chat and completions requests. Since MyScale relies on OpenAI embeddings, it would be good to enable caching functionality for it.

A longer term goal would be to have the OpenAI/custom embeddings function you've added as abstract utils functions outside of MyScale. Feel free to actually add a TODO comment so we can track this!